### DISCUSSION TITLE
Organization Settings and 2FA Requirements

### DISCUSSION THREAD

**Discussion by wagnar-woofbrok** [community/community#34149] [Enterprise] [answered]
Hello,

I'm working on an automations application for my company that involves programmatic access to GitHub via a web crawler. I'm trying to understand the scope of organization settings with regards to 2FA and repositories controlled by a GitHub organization.

Currently, we have 2FA enabled for all accounts. I'm trying to figure out how to allow secure programmatic access to github account credentials and a 2FA code for an automation/bot account.

My questions are:

1. Is 2FA requirement an organization-wide setting only?
2. Can 2FA be scoped to a repository or account regardless of organization settings, or do the organization security/2FA settings override other settings?
3. If I'm specifically trying to take screenshots of a page in GitHub as an authenticated user, is anyone aware of an API or method other than a web crawler logging in with username/password/2FA and browsing to the desired pages?

Thanks for your time!

**Comment by byrneh** [comment, marked as answer]
@wagnar-woofbrok 
w.r.t to GutHub 2FA.
1. Yes, it can also be enforced from enterprise policy (if you have an enteropise account) to all enterprise owned organizations.
2. No 2FA cannot be scoped to the repo being accessed see [requiring-two-factor-authentication-in-your-organization](https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-two-factor-authentication-for-your-organization/requiring-two-factor-authentication-in-your-organization)
3. Taking screenshots, web crawling for programatic and automation is really an anti-patten. Yuu shoud be able to GET all the data you need via REST and GRAPHQL API endpoints. https://docs.github.com/en/rest and https://docs.github.com/en/graphql

**Reply by wagnar-woofbrok** [reply]
Thank you, @byrneh ! That clearly answers my questions, I appreciate it. I'm trying to scope and plan an implementation so knowing my options up front helps.

In regards to web crawling, you're totally right - it's brittle and hard to maintain and not efficient. In my case, I very specifically need screenshots from a web browser with a timestamp because it's one of the requirements I was given. 

Cheers.
